tools/appveyor.psm1

```diff
@@ -141,36 +239,41 @@ function Invoke-AppveyorFinish
         [System.IO.Compression.ZipFile]::CreateFromDirectory($env:CoreOutput, $zipFilePath)
         Write-Verbose "Zipping ${env:FullOutput} into $zipFileFullPath" -verbose
         [System.IO.Compression.ZipFile]::CreateFromDirectory($env:FullOutput, $zipFileFullPath)
-        
+
         $artifacts = New-Object System.Collections.ArrayList
         foreach ($package in $packages) {
             $artifacts.Add($package)
         }
-        
+
         $artifacts.Add($zipFilePath)
         $artifacts.Add($zipFileFullPath)
 
         if ($env:APPVEYOR_REPO_TAG_NAME)
         {
-          # ignore the first part of semver, use the preview part
-          $preReleaseVersion = ($env:APPVEYOR_REPO_TAG_NAME).Split('-')[1]
+            # ignore the first part of semver, use the preview part
+            $preReleaseVersion = ($env:APPVEYOR_REPO_TAG_NAME).Split('-')[1]
         }
         else
         {
-          $previewLabel = (git describe --abbrev=0).Split('-')[1]
-          $preReleaseVersion = "$previewLabel.$($env:APPVEYOR_BUILD_NUMBER)"
+            $previewLabel = (git describe --abbrev=0).Split('-')[1].replace('.','')
+            if(Test-DailyBuild)
+            {
+                $previewLabel= "daily-{0}" -f $previewLabel
+            }
+
+            $preReleaseVersion = "$previewLabel-$($env:APPVEYOR_BUILD_NUMBER.replace('.','-'))"
         }
 
         # only publish to nuget feed if it is not a pull request
-        if(!$env:APPVEYOR_PULL_REQUEST_NUMBER)
+        if(!$env:APPVEYOR_PULL_REQUEST_NUMBER -and $env:TestPassed -eq 'True')
         {
             Publish-NuGetFeed -OutputPath .\nuget-artifacts -VersionSuffix $preReleaseVersion
         }
 
-        $artifacts += (ls .\nuget-artifacts | % {$_.FullName})
+        $artifacts += (Get-ChildItem .\nuget-artifacts | ForEach-Object {$_.FullName})
```

Comment:
talked with Travis offline, we agreed to add `-erroraction silientlycontinue` to `Get-ChildItem` here.